File: Leetcode-Problems/Minimum Deletions to Make Array Divisible.cpp
class Solution
{
public:
    int minOperations(vector<int> &nums, vector<int> &numsDivide)
    {
        int gc = numsDivide[0];
        for (int i = 1; i < numsDivide.size(); i++)
        {
            gc = __gcd(gc, numsDivide[i]);
        }
        int ans = 0;
        sort(nums.begin(), nums.end());
        for (int i : nums)
        {
            if (gc % i == 0)
            {
                return ans;
            }
            ans++;
        }
        return -1;
    }
};

File: Leetcode-Problems/Minimum Deletions to Make Array Divisible.java
class Solution {
    public int gcd(int a, int b) {
        if (b == 0)
            return a;
        return gcd(b, a % b);
    }

    public int minOperations(int[] nums, int[] numsDivide) {
        int res = numsDivide[0];
        for (int i : numsDivide) {
            res = gcd(res, i);
        }
        int count = 0;
        boolean flag = false;
        Arrays.sort(nums);
        for (int i : nums) {
            if (res % i != 0) {
                count++;
            } else {
                flag = true;
                break;
            }
        }
        return flag ? count : -1;
    }
}